src/mavsdk/core/mavlink_mission_transfer.cpp

```diff
@@ -237,11 +237,27 @@ void MavlinkMissionTransfer::UploadWorkItem::start()
         return;
     }
 
-    int count = 0;
-    for (const auto& item : _items) {
-        if (count++ != item.seq) {
-            callback_and_reset(Result::InvalidSequence);
-            return;
+    // Mission items for ArduPilot are off by one because the home position is
+    // item 0, and the sequence starts counting at 0 from 1.
+    // This is only for missions, rally points, and geofence items are normal.
+    const bool one_off =
```

Comment:
Nitpick, trying to keep the ardupilot-specific bits very clear:

```suggestion
    const bool is_ardupilot_mission =
```

Reply:
Much better, thanks!